app/main.py

```diff
@@ -1,6 +1,35 @@
+import os
+import requests
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+API_KEY = os.getenv("API_KEY")
+URL = "https://api.weatherapi.com/v1/current.json"
+CITY = "Paris"
+
+
 def get_weather() -> None:
-    # write your code here
-    pass
+    print(f"Start request to Weather API for city {CITY}...")
+    res = requests.get(URL, params={"key": API_KEY, "q": CITY})
+
+    if res.status_code == 200:
+        data = res.json()
+        location_name = data["location"]["name"]
+        country = data["location"]["country"]
+        local_time = data["location"]["localtime"]
+        temp_c = data["current"]["temp_c"]
+        weather = data["current"]["condition"]["text"]
+
+        print(
+            f"{location_name}/{country} "
+            f"{local_time} Weather: {temp_c} Celsius, {weather}"
+        )
+    else:
+        print(
+            f"status code {res.status_code} something went wrong {res.reason}"
+        )
```

Comment:
Why Use a try Block Here?
Network Errors: The requests.get call can fail due to network issues, timeouts, or invalid URLs. Wrapping it in a try block ensures that these errors are caught and handled gracefully rather than causing your program to crash.

HTTP Errors: If the API returns an error response (e.g., 4xx or 5xx status codes), using response.raise_for_status() will raise an exception. This exception is caught by the except block to provide a meaningful error message.

JSON Parsing Errors: The response.json() call can fail if the response is not valid JSON. Wrapping this in a try block allows you to catch ValueError and handle it appropriately.

Missing Keys: The code assumes that certain keys will always be present in the JSON response. If the API response format changes or if the data is missing, this can raise a KeyError, which you can catch and handle to avoid crashes.

Unforeseen Errors: Catching a generic Exception ensures that any other unexpected errors are also handled, providing robustness to your function.